rs-backend/src/api.rs

```diff
@@ -0,0 +1,167 @@
+// Types implement both serialize and deserialize despite conceptually needing
+// only one for testing.
+
```

Comment:
Would be good to have a doc comment explaining the one mod per API request layout, etc.